### ISSUE TITLE
Plotting a model (with model_to_dot) fails if the inputlabels/outputlabels contain brackets

### ISSUE THREAD

**Issue by cBournhonesque** [keras-team/tf-keras#647]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Macos BigSur
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.7
- Python version: 3.8
- Bazel version (if compiling from source): 
- GPU model and memory:
- Exact command to reproduce:
```
tf.keras.utils.plot_model(model,
                              to_file='model_dir/model.png'),
                              show_shapes=True,
                              show_dtype=True,
                              show_layer_names=True)
```

**Describe the problem**.

When I try to plot a model that contains layers with dictionary inputs, I get an error in the 
`model_to_dot` function in `vis_utils.py`
that says: `Error: invalid label format`.
This error comes from having an invalid graphviz label name (defined here: https://github.com/keras-team/keras/blob/master/keras/utils/vis_utils.py#L293)
My input shape is a dictionary, and is in the form: `{'a': (None, 1), 'b': (None, 2)}`. 
If I call `plot_model` with `show_shapes=True`, then the shape will be added to the label name for graphbiz.
The problem is that the brackets in the `inputlabels` and `outputlabels` need to be escaped so that the node's label can be interpreted correctly by graphviz. (otherwise, graphviz interprets it as a nested label: https://graphviz.org/doc/info/shapes.html#record)

I fixed the issue by adding:
```
inputlabels = inputlabels.replace('{', '\{')
inputlabels = inputlabels.replace('}', '\}')
outputlabels = outputlabels.replace('{', '\{')
outputlabels = outputlabels.replace('}', '\}')
```
before https://github.com/keras-team/keras/blob/master/keras/utils/vis_utils.py#L293
but there must be a more elegant way of fixing this.

**Describe the current behavior**.
Plotting the model fails.

**Describe the expected behavior**.
The model gets plotted correctly

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):
Maybe we could have a list of characters to escape ('{', '}', '|', etc.), and escape all of these at once in the label to be plotted?
